### PR TITLE
_brand.yml: fonts in `html`-like formats, font colors, schema updates, etc

### PR DESCRIPTION
**Pull request by cscheid**
Towards https://github.com/quarto-dev/quarto-cli/issues/10249.

This adds support for `typography` entries in `html` formats.

Notably, we don't currently download the fonts. I'm not sure it's something we want to do or not in HTML formats (it will be necessary for google font support in `typst`)

### CODE REVIEW

File: src/resources/schema/definitions.yml [hunk 1]
```diff
@@ -2792,7 +2793,30 @@
...
+      200,
+      300,
+      400,
+      500,
+      600,
+      700,
+      800,
+      900,
+      thin,
+      extra-light,
```
Comment (gadenbuie) [inline src/resources/schema/definitions.yml:2808]:
```suggestion
      extra-light,
      ultra-light,
```

**Comment by gordonwoodhull** [comment]
I've updated the typst brand yaml implementation and tests to the new schema.

Remaining failures don't seem to be related to brand yaml.

I think this is fine to merge. I'll continue with caching of fonts for typst, and hopefully have that ready for review at eng meeting tomorrow.

**Comment by gordonwoodhull** [comment]
I've disabled font bunny on the Typst side for now (and removed the failing test), but we can pick it back up if we find a better woff->ttf converter.

I think we should merge this.

**Comment by cscheid** [comment]
I cleaned up all this work into two main commits, one for the SCSS->CSS variable/SCSS analyzer work, and the other for the typography work in typst+html formats.

**Pull request merged**